src/test/java/teammates/test/cases/browsertests/FeedbackMcqQuestionUiTest.java

```diff
@@ -291,4 +292,177 @@ public void testDeleteQuestionAction() {
+        assertNotNull(BackDoor.getFeedbackQuestion(courseId, feedbackSessionName, 1));
+
+        // verify html page
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackMcqQuestionWeightAddSuccess.html");
+
+        ______TS("MCQ: Test front-end validation for empty weights");
+        feedbackEditPage.clickEditQuestionButton(1);
+
+        // Check validation for MCQ weight cells
+        feedbackEditPage.fillMcqWeightBox(1, 0, "");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.isMcqWeightBoxFocused(1, 0);
+        feedbackEditPage.fillMcqWeightBox(1, 0, "0");
+
+        // Check validation for empty other weight
+        feedbackEditPage.fillMcqOtherWeightBox(1, "");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.isMcqOtherWeightBoxFocused(1);
+        feedbackEditPage.fillMcqOtherWeightBox(1, "0");
+
+        // Check validation for the weight cells added by 'add option' button
+        feedbackEditPage.clickAddMoreMcqOptionLink(1);
+        feedbackEditPage.isElementPresent(By.id("mcqWeight-2-1"));
+        feedbackEditPage.fillMcqOption(1, 2, "Choice 3");
+        feedbackEditPage.fillMcqWeightBox(1, 2, "");
+        feedbackEditPage.clickAddQuestionButton();
+        feedbackEditPage.isMcqWeightBoxFocused(1, 2);
+        feedbackEditPage.clickDiscardChangesLink(1);
+        feedbackEditPage.waitForConfirmationModalAndClickOk();
+
+        ______TS("MCQ: Edit weights success");
+        feedbackEditPage.clickEditQuestionButton(1);
+        feedbackEditPage.fillMcqWeightBox(1, 0, "1.55");
+        feedbackEditPage.fillMcqWeightBox(1, 1, "2.77");
+        feedbackEditPage.fillMcqOtherWeightBox(1, "3.66");
+        feedbackEditPage.clickSaveExistingQuestionButton(1);
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_EDITED);
+
+        // verify weights for the question
+        feedbackEditPage.verifyFieldValue("mcqWeight-0-1", "1.55");
+        feedbackEditPage.verifyFieldValue("mcqWeight-1-1", "2.77");
+        feedbackEditPage.verifyFieldValue("mcqOtherWeight-1", "3.66");
+
+        // verify html page
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackMcqQuestionWeightEditSuccess.html");
+
+        ______TS("MCQ: Add more weights");
+        feedbackEditPage.clickEditQuestionButton(1);
+        feedbackEditPage.clickAddMoreMcqOptionLink(1);
+        assertTrue(feedbackEditPage.isElementPresent("mcqOption-2-1"));
+        assertTrue(feedbackEditPage.isElementPresent("mcqWeight-2-1"));
+        feedbackEditPage.fillMcqOption(1, 2, "Choice 3");
+        feedbackEditPage.fillMcqWeightBox(1, 2, "4");
+        feedbackEditPage.clickSaveExistingQuestionButton(1);
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_EDITED);
+
+        ______TS("MCQ: Test weight is not added for invalid choice value");
```

Comment:
The case is a bit "redundant" as u already has the same scenario in the integration/unit test. You can just have one failure case and that should be enough.

Reply:
>You can just have one failure case and that should be enough.

@xpdavid By this, do you mean that the failure tests I added in unit test is enough and I should delete this test? Or, do you mean, I should have only one failure case in <em>this</em> test only. (That is, to keep either the `"                "` (space only) choice or the empty choice case)?

Reply:
> do you mean that the failure tests I added in unit test is enough
Yes. Validate all scenarios in the unit tests should be enough.

You don't have to delete them I think as you have put lots of efforts in coming out and writing them.